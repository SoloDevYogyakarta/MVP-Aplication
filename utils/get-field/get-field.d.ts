import { GetFieldType } from 'src/types/get-field.type';
export declare function getField(filename?: GetFieldType): any;
