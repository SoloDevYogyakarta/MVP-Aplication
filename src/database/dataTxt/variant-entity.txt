{"id":75,"name":"rantai","type":"sss","desc":"bawa sendiri","product_id":"jdJ_bHVhSPV35O5IgcnDn","public_id":"X6_vVxjbTTBTMHX0_I5P2"}