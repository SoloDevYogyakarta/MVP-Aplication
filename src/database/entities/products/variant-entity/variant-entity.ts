import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class VariantEntity extends BaseEntity {
  name: string;
  type: string;
  desc: string;
  price: number;
  product_id: string;
}

export type VariantCreationAttribute = Optional<VariantEntity, 'id'>;
export interface VariantInstance
  extends Model<VariantCreationAttribute, VariantEntity>,
    VariantEntity {}
export const variantEntity = sequelize.define<VariantInstance>(
  'VARIANT',
  {
    public_id: DataTypes.STRING,
    name: DataTypes.STRING,
    type: DataTypes.STRING,
    desc: DataTypes.STRING,
    price: DataTypes.INTEGER,
    product_id: DataTypes.STRING,
  } as ModelAttributes<VariantInstance>,
  {
    tableName: 'VARIANT',
    schema: 'PRODUCTS',
    modelName: 'variant',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
