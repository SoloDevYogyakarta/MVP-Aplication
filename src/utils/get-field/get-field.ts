import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
