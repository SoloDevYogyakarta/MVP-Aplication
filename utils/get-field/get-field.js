"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getField = void 0;
const system_1 = require("../system/system");
function getField(filename) {
    try {
        const inc = ['token', 'fake'];
        if (inc.includes(filename)) {
            return (0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`);
        }
        return JSON.parse((0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`));
    }
    catch (err) {
        throw new Error(err);
    }
}
exports.getField = getField;
//# sourceMappingURL=get-field.js.map