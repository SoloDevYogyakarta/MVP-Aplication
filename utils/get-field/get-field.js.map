{"version":3,"file":"get-field.js","sourceRoot":"","sources":["../../../src/utils/get-field/get-field.ts"],"names":[],"mappings":";;;AACA,6CAA4C;AAE5C,SAAgB,QAAQ,CAAC,QAAuB;IAC9C,IAAI;QACF,MAAM,GAAG,GAAG;YACV,MAAM;YACN,OAAO;YACP,cAAc;YACd,cAAc;YACd,gBAAgB;SACjB,CAAC;QACF,IAAI,GAAG,CAAC,QAAQ,CAAC,QAAQ,CAAC,EAAE;YAC1B,OAAO,IAAA,iBAAQ,EAAC,kBAAkB,QAAQ,MAAM,CAAC,CAAC;SACnD;QACD,OAAO,IAAI,CAAC,KAAK,CAAC,IAAA,iBAAQ,EAAC,kBAAkB,QAAQ,MAAM,CAAC,CAAC,CAAC;KAC/D;IAAC,OAAO,GAAG,EAAE;QACZ,MAAM,IAAI,KAAK,CAAC,GAAG,CAAC,CAAC;KACtB;AACH,CAAC;AAhBD,4BAgBC"}