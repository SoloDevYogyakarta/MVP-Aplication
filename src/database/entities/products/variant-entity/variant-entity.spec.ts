import { ModelCtor } from 'sequelize';
import { getField } from '../../../../utils/get-field/get-field';
import { variantEntity, VariantInstance } from './variant-entity';

describe('variantEntity', () => {
  let entity: ModelCtor<VariantInstance>;

  beforeEach(async () => {
    entity = await variantEntity;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  try {
    it('findOne', async () => {
      const { public_id } = getField('variant-entity');
      const findOne = await entity.findOne({ where: { public_id } });
      expect(findOne.public_id).toEqual(public_id);
    });

    it('findAll', async () => {
      const findAll = await entity.findAll();
      expect(findAll.length).not.toEqual(0);
    });
  } catch (err) {
    // empty
  }
});
