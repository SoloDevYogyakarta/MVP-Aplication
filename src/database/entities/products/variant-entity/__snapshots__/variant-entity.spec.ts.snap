// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variantEntity render correctly 1`] = `[Function]`;
